zcash_primitives/src/local_consensus.rs

```diff
@@ -0,0 +1,284 @@
+use crate::{
+    consensus::{BlockHeight, Network, NetworkUpgrade, Parameters},
+    constants,
+};
+
+/// Represents the Network Parameters of the Full Node the caller
+/// would be connecting to.
+/// `Consensus(Network::MainNetwork)` refers to a Zcash Full Node
+/// running on public Mainnet
+/// `Consensus(Network::TestNetwork)` refers to a Zcash Full Node
+/// running on public Testnet
+/// `Local(P)` can be used to define the consensus parameters of
+/// a local or non-public network like one or more nodes on
+/// `regtest` mode.
+pub enum FullNodeParameters<P: Parameters> {
+    Consensus(Network),
+    Local(P),
+}
```

Comment:
```suggestion
/// Represents the Network Parameters of the Full Node the caller connects to.
pub enum FullNodeParameters<P: Parameters> {
    /// Variant indicating a Zcash Full Node running on public Mainnet or Testnet
    Consensus(Network),
    /// The consensus parameters of a local or non-public network like one or more nodes 
    /// in `regtest` mode.
    Local(P),
}
```

Who is "the caller" here?

Reply:
how do you refer to those who are using librustzcash in the documentation?

Reply:
Hm, actually, `FullNodeParameters` is a bit misleading; this is basically just `NetworkParameters` or something. It doesn't really pertain to full nodes specifically, just whatever network the client is connecting to. 

I'm now actually a little puzzled why a concrete enum is needed for this type at all; since everything is bounded on `consensus::Parameters`, don't you just create a new concrete type for your network parameters and use that in the caller? Under what circumstances do you need to switch between parameters at runtime (as this enum lets you do)?

Reply:
Going back in the conversation of this change the original idea was to follow the current enum approach were one would specify MainNetwork or TestNetwork and add a Regtest, that didn't make much sense because Regtest is not consensus but a particular Node setup. So FullNodeParameters came up as a way to still group the possible node setups a client could connect to without changing the existing API fo `consensus::`.

Particularly, the SDKs' `RustBackend` would receive this FullNodeParameters that could be `Consensus` or `Local` (regtest or other kind of setup with different activation heights, etc.). What you suggest is another way of achieving the same. The mobile SDKs would need update anyway so we could do that as well.